### PR TITLE
Mark as unmaintained

### LINKED ISSUE #119: Mark this repository as unmaintained
Sadly we haven't got the time to maintain this repository as an official GitHub Action. In the last year or two since we originally published this, a large number of really good community Actions have appeared on the Marketplace for doing this.

We'd like to mark this repository as unmaintained and make it clear to users what their expectations should be.

People are of course free to fork this repository and maintain those forks if they like.

To do this we'll:
* [x] Remove it from the Marketplace 
* [x] Remove from any official GitHub documentation.
* [x] Change the README.md to clearly reflect its new state.
* [x] Update the README.md to point to other community Actions.
* [x] Archive this repository, putting it read-only mode. It will continue to work for all existing Action users, but won't receive any updates.

### PR DESCRIPTION
**Pull request by andymckay**
Add in a notice about the upcoming lack of intended maintenance on this repository. We'd like to make it clear to visitors to this repository about it's state. This process of marking it unmaintained only affects future updates. The Action will still continue work as it is in old workflows.

Closes #119

Once this lands, we'll also Archive this repository.

If you've got good recommendations for replacements, please let me know, we'll add them into the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,20 @@
 # GitHub Action - Releases API
+
+**Please note:** This repository is currently unmaintained by a team of developers at GitHub. The 
+repository is here and you can use it as an example, or in Actions. However please be aware that 
+we are not going to be updating issues or pull requests on this repository.
+
+**Maintained forks:**
+* ...
```
Comment (andymckay) [inline README.md:8]: Please let me know about good alternatives, we'll add them in here.

**Comment by Margen67** [comment]
Alternatives:
  * https://github.com/marvinpinto/action-automatic-releases
  * https://github.com/softprops/action-gh-release

File: README.md (hunk 1 above)
Comment (andymckay) [inline README.md]:
```suggestion
* https://github.com/marvinpinto/action-automatic-releases
* https://github.com/softprops/action-gh-release
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,20 @@
 # GitHub Action - Releases API
+
+**Please note:** This repository is currently unmaintained by a team of developers at GitHub. The 
+repository is here and you can use it as an example, or in Actions. However please be aware that 
+we are not going to be updating issues or pull requests on this repository.
+
+**Maintained Actions:**
+* [elgohr/Github-Release-Action](https://github.com/elgohr/Github-Release-Action)
```
Comment (andymckay) [inline README.md:8]:
```suggestion
* [elgohr/Github-Release-Action](https://github.com/elgohr/Github-Release-Action)
* [marvinpinto/action-automatic-releases](https://github.com/marvinpinto/action-automatic-releases)
* [softprops/action-gh-release](https://github.com/softprops/action-gh-release)
* [ncipollo/release-action](https://github.com/ncipollo/release-action)

```

**Comment by andymckay** [comment]
Thanks for the updates in the pull request and issue, I've added them here.

**Pull request merged**